extensions/ql-vscode/test/vscode-tests/no-workspace/model-editor/model-editor-view.test.ts

```diff
@@ -0,0 +1,79 @@
+import { join } from "path";
+import { Mode } from "../../../../src/model-editor/shared/mode";
+import { mockDatabaseItem, mockedObject } from "../../utils/mocking.helpers";
+import { CodeQLCliServer } from "../../../../src/codeql-cli/cli";
+import { Uri, WebviewPanel } from "vscode";
+import { ModelEditorView } from "../../../../src/model-editor/model-editor-view";
+import { createMockApp } from "../../../__mocks__/appMock";
+import { mockEmptyDatabaseManager } from "../query-testing/test-runner-helpers";
+import { QueryRunner } from "../../../../src/query-server";
+import { ExtensionPack } from "../../../../src/model-editor/shared/extension-pack";
+
+describe("ModelEditorView", () => {
+  const app = createMockApp({});
+  const databaseManager = mockEmptyDatabaseManager();
+  const cliServer = mockedObject<CodeQLCliServer>({});
+  const queryRunner = mockedObject<QueryRunner>({});
+  const queryStorageDir = "/a/b/c/d";
+  const queryDir = "/a/b/c/e";
+  const databaseItem = mockDatabaseItem();
+  const extensionPack: ExtensionPack = {
+    path: "/a/b/c/f",
+    yamlPath: join("/a/b/c/f", "codeql-pack.yml"),
+    name: "codeql/test",
+    version: "0.0.0",
+    language: "java",
+    extensionTargets: {
+      "codeql/java-all": "*",
+    },
+    dataExtensions: ["models/**/*.yml"],
+  };
+  const mode = Mode.Application;
+  const updateMethodsUsagePanelState = jest.fn();
+  const showMethod = jest.fn();
+  const handleViewBecameActive = jest.fn();
+  const handleViewWasDisposed = jest.fn();
+  const isMostRecentlyActiveView = jest.fn();
+
+  let view: ModelEditorView;
+
+  beforeEach(() => {
```

Comment:
Should we add another test that calls `view.openView()`?

Reply:
Oops, I missed that in the internal issue you said it probably isn't worth doing any more tests here.

Reply:
I've added a comment to the test why we're using `restoreView` rather than `openView`. I don't think there's a lot of interesting logic in `openView` that is not also tested by using `restoreView`; we're mostly testing whether `getPanelConfig` returns the correct things.

Reply:
Thanks for explaining and adding the comment in the code.